src/routes/(app)/c/[channel_name]/+page.server.ts

```diff
@@ -0,0 +1,26 @@
+import { getDb } from '$lib/server';
+import { getPublicChannelByName } from '$lib/server/services/channels';
+import { getPosts } from '$lib/server/services/content';
+import { error } from '@sveltejs/kit';
+import type { PageServerLoad } from './$types';
+
+export const load = (async ({ params: { channel_name }, locals }) => {
+    const db = await getDb();
+    const channel = await getPublicChannelByName(db, channel_name);
+
+    if (!channel) {
+        error(404);
```

Comment:
It'd be nice to provide an error message here.

Something like `Channel of name ${channel_name} not found` would suffice.